integration/teardown/teardown.go

```diff
@@ -3,40 +3,143 @@
 package teardown
 
 import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/backoff"
 	"github.com/giantswarm/e2e-harness/pkg/framework"
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	"github.com/giantswarm/kvm-operator/integration/env"
+	"github.com/giantswarm/kvm-operator/integration/utils"
 )
 
 // Teardown e2e testing environment.
 func Teardown(g *framework.Guest, h *framework.Host) error {
 	var err error
+	var l micrologger.Logger
+	{
+		c := micrologger.Config{}
+
+		l, err = micrologger.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+	clusterID := getClusterID(h.TargetNamespace())
+	ctx := context.Background()
 
+	// get flannel info so we can delete it from rangepool
+	var flannelNetwork string
 	{
-		// only do full teardown when not on CI
-		if env.CircleCI() == "true" {
-			return nil
+		flannelConfig, err := h.G8sClient().CoreV1alpha1().FlannelConfigs(v1.NamespaceDefault).Get(clusterID, v1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
 		}
+		flannelNetwork = flannelConfig.Spec.Flannel.Spec.Network
 	}
 
 	{
-		err = framework.HelmCmd("delete kvm-operator --purge")
+		err = framework.HelmCmd(fmt.Sprintf("delete %s-cert-config-e2e --purge", h.TargetNamespace()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		err = framework.HelmCmd(fmt.Sprintf("delete %s-flannel-config-e2e --purge", h.TargetNamespace()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		err = framework.HelmCmd(fmt.Sprintf("delete %s-kvm-config-e2e --purge", h.TargetNamespace()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// wait until crds are deleted by operators
+	o := func() error {
+		certList, err := h.G8sClient().CoreV1alpha1().CertConfigs(v1.NamespaceDefault).List(v1.ListOptions{
+			LabelSelector: crdLabelSelector(clusterID),
+		})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(certList.Items) == 0 {
+			// resource doesnt exist, we are good to continue
+			l.Log("level", "info", "message", "cert crd was deleted")
```

Comment:
We seem to have a context available, so all logging should make use of it via `LogCtx`.

Reply:
fixed